doc/neps/nep-0031-uarray.rst

```diff
@@ -69,12 +70,37 @@ _will be discussed on the mailing list._
 
 The way we propose the overrides will be used by end users is::
 
-    from numpy import unumpy
-    TODO
+    import numpy.overridable as np
+    with np.set_backend(backend):
+        x = np.asarray(my_array, dtype=dtype)
 
-And a library that implements a NumPy-like API will use it like::
+And a library that implements a NumPy-like API will use it in the following manner (as an example)::
 
-    TODO: example corresponding to NEP 30 `duckarray`
+    import numpy.overridable as np
+    _ua_implementations = {}
+
+    __ua_domain__ = "numpy"
+
+    def __ua_function__(func, args, kwargs):
+        return _ua_implementations[func](*args, **kwargs)
+
+    def implements(ua_func):
+        def inner(func):
+            _ua_implementations[ua_func] = func
+            return func
+
+        return inner
+
+    @implements(np.asarray)
+    def asarray(a, dtype=None, order=None):
+        # Code here
+        # Must return NotImplemented for unsupported array types
+
+    # Assuming your arrays are mutable, this also implements
+    # zeros and ones.
```

Comment:
why the disclaimer about mutability? this is going to raise questions again, let's leave this out. None of the libraries that care about this only provide immutable arrays anyway I'd think